nixos/modules/services/web-servers/nginx/default.nix

```diff
@@ -638,17 +652,34 @@ in
       preStart =
         ''
         ${cfg.preStart}
-        ${cfg.package}/bin/nginx -c ${configFile} -p ${cfg.stateDir} -t
+        ${cfg.package}/bin/nginx -c ${configPath} -p ${cfg.stateDir} -t
         '';
       serviceConfig = {
-        ExecStart = "${cfg.package}/bin/nginx -c ${configFile} -p ${cfg.stateDir}";
+        ExecStart = "${cfg.package}/bin/nginx -c ${configPath} -p ${cfg.stateDir}";
         ExecReload = "${pkgs.coreutils}/bin/kill -HUP $MAINPID";
         Restart = "always";
         RestartSec = "10s";
         StartLimitInterval = "1min";
       };
     };
 
+    environment.etc."nginx/nginx.conf" = mkIf cfg.enableReload {
```

Comment:
Hm, why is this only enabled if `enableReload` is `true`? Apart from that, I do have mixed feelings about exposing the config in `/etc`, because I do not want other applications to make certain assumptions about files `/etc` and generally want to keep as much things out of global paths to ensure they don't behave differently in a Nix build sandbox. That said, I can still `mkForce` it to an empty attrset, but maybe adding `exposeConfigFile` with a default of `false` might be a good idea? Maybe depends on how many users want it to occur in `/etc` versus how many do not want that behaviour.

Reply:
ohhh, the original commit actually called that right https://github.com/NixOS/nixpkgs/pull/57429/commits/8b036d54acd898c3ac2bd8b8ad734c83ff6f8179#diff-795fa65a7c41a479084de826e4e2e65cR460

It just people suggested that `exposeConfig` isn't a good name and it should actually be `enableReload`

Actually, I'd like to write that file in some other place, but only `/etc` has atomic replaces in NixOS. I don't want to move `nginx.conf` to runtime dir, as I'd like to have post-mortem (after nginx doesn't start) config checks.